test/infrastructure/docker/exp/internal/controllers/dockermachinepool_controller.go

```diff
@@ -380,6 +375,63 @@ func dockerMachineToDockerMachinePool(_ context.Context, o client.Object) []ctrl
+func (r *DockerMachinePoolReconciler) updateStatus(ctx context.Context, cluster *clusterv1.Cluster, machinePool *expv1.MachinePool, dockerMachinePool *infraexpv1.DockerMachinePool, dockerMachines []infrav1.DockerMachine) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	// List the Docker containers. This corresponds to an InfraMachinePool instance for providers.
+	labelFilters := map[string]string{dockerMachinePoolLabel: dockerMachinePool.Name}
+	externalMachines, err := docker.ListMachinesByCluster(ctx, cluster, labelFilters)
+	if err != nil {
+		return ctrl.Result{}, errors.Wrapf(err, "failed to list all machines in the cluster")
+	}
+
+	externalMachineMap := make(map[string]*docker.Machine)
+	for _, externalMachine := range externalMachines {
+		externalMachineMap[externalMachine.Name()] = externalMachine
+	}
+	_, readyMachines, err := r.getDeletionCandidates(ctx, dockerMachines, externalMachineMap, machinePool, dockerMachinePool)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	readyReplicaCount := len(readyMachines)
+	desiredReplicas := int(*machinePool.Spec.Replicas)
+
+	switch {
+	// We are scaling up
+	case readyReplicaCount < desiredReplicas:
+		conditions.MarkFalse(dockerMachinePool, clusterv1.ResizedCondition, clusterv1.ScalingUpReason, clusterv1.ConditionSeverityWarning, "Scaling up MachineSet to %d replicas (actual %d)", desiredReplicas, readyReplicaCount)
+	// We are scaling down
+	case readyReplicaCount > desiredReplicas:
+		conditions.MarkFalse(dockerMachinePool, clusterv1.ResizedCondition, clusterv1.ScalingDownReason, clusterv1.ConditionSeverityWarning, "Scaling down MachineSet to %d replicas (actual %d)", desiredReplicas, readyReplicaCount)
+	default:
+		// Make sure last resize operation is marked as completed.
+		// NOTE: we are checking the number of machines ready so we report resize completed only when the machines
+		// are actually provisioned (vs reporting completed immediately after the last machine object is created). This convention is also used by KCP.
+		if len(dockerMachines) == readyReplicaCount {
+			if conditions.IsFalse(dockerMachinePool, clusterv1.ResizedCondition) {
+				log.Info("All the replicas are ready", "replicas", readyReplicaCount)
+			}
+			conditions.MarkTrue(dockerMachinePool, clusterv1.ResizedCondition)
+		}
+		// This means that there was no error in generating the desired number of machine objects
+		conditions.MarkTrue(dockerMachinePool, clusterv1.MachinesCreatedCondition)
+	}
+
+	getters := make([]conditions.Getter, 0, len(dockerMachines))
+	for i := range dockerMachines {
+		getters = append(getters, &dockerMachines[i])
+	}
+
+	// Aggregate the operational state of all the machines; while aggregating we are adding the
+	// source ref (reason@machine/name) so the problem can be easily tracked down to its source machine.
+	conditions.SetAggregate(dockerMachinePool, expv1.ReplicasReadyCondition, getters, conditions.AddSourceRef(), conditions.WithStepCounterIf(false))
```

Comment:
conditions.WithStepCounterIf(false) doesn't seem to do anything as we always pass in false (?)

Reply:
If I'm right, let's get rid of all other occurences of this (ideally in a separate PR I think)(

Reply:
There are a lot of occurrences of this, let's change all of them in a separate PR

https://github.com/kubernetes-sigs/cluster-api/issues/9754